autora/cycle/plot_utils.py

```diff
@@ -0,0 +1,470 @@
+        "s": 2,
+        "alpha": 0.6,
+        "label": "New Data",
+    }
+    line_kw_defaults = {"label": "Theory"}
+    # Combine default and user supplied keywords
+    d_kw = {}
+    for d1, d2, key in zip(
+        [
+            subplot_kw_defaults,
+            scatter_kw1_defaults,
+            scatter_kw2_defaults,
+            line_kw_defaults,
+        ],
+        [subplot_kw, scatter_kw1, scatter_kw2, line_kw],
+        ["subplot_kw", "scatter_kw1", "scatter_kw2", "line_kw"],
+    ):
+        assert isinstance(d1, dict)
+        assert isinstance(d2, dict)
+        d_kw[key] = check_replace_default_kw(d1, d2)
+
+    # ---Extract IVs and DV metadata and indexes---
+    ivs, dvs = get_variable_index(cycle)
+    if iv_name:
+        iv = [s for s in ivs if s[1] == iv_name][0]
+    else:
+        iv = [ivs[0]][0]
+    if dv_name:
+        dv = [s for s in dvs if s[1] == dv_name][0]
+    else:
+        dv = [dvs[0]][0]
+    iv_label = f"{iv[1]} {iv[2]}"
+    dv_label = f"{dv[1]} {dv[2]}"
+
+    # Create a dataframe of observed data from cycle
+    df_observed = observed_to_df(cycle)
+
+    # Generate IV space
+    condition_space = generate_condition_space(cycle, steps=steps)
+    # Get theory predictions over space
+    d_predictions = theory_predict(cycle, condition_space)
+
+    # Subplot configurations
+    if n_cycles < wrap:
+        shape = (1, n_cycles)
+    else:
+        shape = (int(np.ceil(n_cycles / wrap)), wrap)
+    fig, axs = plt.subplots(*shape, **d_kw["subplot_kw"])
+
+    # Loop by panel
+    for i, ax in enumerate(axs.flat):
+        if i + 1 <= n_cycles:
+
+            # ---Plot observed data---
+            # Independent variable values
+            x_vals = df_observed.loc[:, iv[0]]
+            # Dependent values masked by current cycle vs previous data
+            dv_previous = np.ma.masked_where(
+                df_observed["cycle"] >= i, df_observed[dv[0]]
+            )
+            dv_current = np.ma.masked_where(
+                df_observed["cycle"] != i, df_observed[dv[0]]
+            )
+            # Plotting scatter
+            ax.scatter(x_vals, dv_previous, **d_kw["scatter_kw1"])
+            ax.scatter(x_vals, dv_current, **d_kw["scatter_kw2"])
+
+            # ---Plot Theory---
+            conditions = condition_space[:, iv[0]]
+            ax.plot(conditions, d_predictions[i], **d_kw["line_kw"])
+
+            # Label Panels
+            ax.text(0.05, 1, f"Cycle {i}", ha="left", va="top", transform=ax.transAxes)
+
+            if not spines:
+                ax.spines.right.set_visible(False)
+                ax.spines.top.set_visible(False)
+
+        else:
+            ax.axis("off")
+
+    # Super Labels
+    fig.supxlabel(iv_label, y=0.07)
+    fig.supylabel(dv_label)
+
+    # Legend
+    fig.legend(
+        ["Previous Data", "New Data", "Theory"],
+        ncols=3,
+        bbox_to_anchor=(0.5, 0),
+        loc="lower center",
+    )
+
+    return fig
+
+
+def plot_results_panel_3d(
```

Comment:
All the same comments apply for this one as for plot_results_panel_2d. 

Is there an example of usage? I saw the 2d one, but not a 3d one.

Reply:
I updated the notebook with a 3D example!